x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -657,4 +664,35 @@ private static void checkMatchCommand(LogicalPlan plan, Set<Failure> failures) {
             }
         }
     }
+
+    private static void checkFullTextQueryFunctions(LogicalPlan plan, Set<Failure> failures) {
+        if (plan instanceof Filter f) {
+            Expression condition = f.condition();
+            if (condition instanceof FullTextFunction ftf) {
+                // Similar to cases present in org.elasticsearch.xpack.esql.optimizer.rules.PushDownAndCombineFilters -
+                // we can't check if it can be pushed down as we don't have yet information about the fields present in the
+                // StringQueryPredicate
+                plan.forEachDown(LogicalPlan.class, lp -> {
+                    if ((lp instanceof Filter || lp instanceof OrderBy || lp instanceof EsRelation) == false) {
+                        failures.add(
+                            fail(
+                                plan,
+                                "Full text functions cannot be used after {}",
+                                lp.sourceText().split(" ")[0].toUpperCase(Locale.ROOT)
+                            )
+                        );
+                    }
+                });
+                if (ftf.query().foldable() == false) {
```

Comment:
The normal/usual way to validate foldable and input data types are in the `resolveType()` method of the function itself, to make it more manageable. Please refer to one of the ES|QL string functions, e.g. `Concat`.

Reply:
Thanks for the pointer! Done in db1da96b92e06a16b780aa6d26df343149798c02